### PR TITLE
fix: Refresh model after speaker deletion

### LINKED ISSUE #5435: Speaker: Deleting Speaker from Dashboard does not remove the deleted speaker from the page
**Describe the bug**

On deleting the speaker from dashboard, the deleted speaker is still shown unless we manually reload the page.

**To Reproduce**
1. Go to Manage Event and then Speaker
2. Click on delete option of any speaker

**Expected behavior**

On deleting the speaker the page should show same behavior as on deleting the session i.e the speaker should be removed without reloading the page manually.

**Desktop (please complete the following information):**
All

**Smartphone (please complete the following information):**
All

### PR DESCRIPTION
**Pull request by akash2408**
Fixes #5435 
#### Short description of what this resolves:

ezgif com-gif-maker

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/events/view/speakers/list.js [hunk 1]
```diff
@@ -74,14 +74,17 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
...
     const speaker =  this.store.peekRecord('speaker', speaker_id, { backgroundReload: true });
-    try {
-      await speaker.destroyRecord();
-      this.notify.success(this.l10n.t('Speaker has been deleted successfully.'));
-    } catch (e) {
-      console.warn(e);
-      this.notify.error(this.l10n.t('An unexpected error has occurred.'));
-    }
-    this.set('isLoading', false);
+    speaker.destroyRecord()
```
Comment (iamareebjamal) [inline app/controllers/events/view/speakers/list.js:77]: Use async await

File: app/controllers/events/view/speakers/list.js [hunk 2]
```diff
@@ -74,17 +74,15 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
...
-      })
-      .finally(() => {
-        this.set('isLoading', false);
-      });
+    try {
+      await speaker.destroyRecord();
+      this.notify.success(this.l10n.t('Speaker has been deleted successfully.'));
+      this.refreshModel.bind(this)();
+    } catch (e) {
+      console.warn(e);
```
Comment (iamareebjamal) [inline app/controllers/events/view/speakers/list.js:82]:
```suggestion
      console.error('Error while deleting speaker', e);
```

File: app/controllers/events/view/speakers/list.js [hunk 3]
```diff
@@ -74,17 +74,15 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
...
-      .catch(() => {
-        this.notify.error(this.l10n.t('An unexpected error has occurred.'));
-      })
-      .finally(() => {
-        this.set('isLoading', false);
-      });
+    try {
+      await speaker.destroyRecord();
+      this.notify.success(this.l10n.t('Speaker has been deleted successfully.'));
+      this.refreshModel.bind(this)();
```
Comment (iamareebjamal) [inline app/controllers/events/view/speakers/list.js:80]: No need for bind call. `this` is literally `this` already

**Comment by iamareebjamal** [comment]
Great. See how simple the PR became. Please add a GIF to show the change

**Comment by akash2408** [comment]
Sure wait a minute

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**